### PR TITLE
fix phpstan level 4 warnings

### PR DESCRIPTION
**Pull request by clxmstaab**
- [ ] Bug fix #…?
- [ ] New feature?
- [ ] BC breaks?
- [ ] Deprecations?
- [ ] Tests added?
- [ ] Changelog updated?

      Please, update CHANGELOG.md by running next command:
      $ php bin/changelog

like in previous PRs with the same topic, this changes don't have a CHANGELOG.md line. this was your prefered way to handle this in the past.

### CODE REVIEW

File: src/functions.php [hunk 1]
```diff
@@ -192,7 +192,7 @@ function task(string $name, $body = null): Task
  * @param string $task The task before $that should be run.
  * @param string|callable $do The task to be run.
  *
- * @return Task|void
```
Comment (clxmstaab) [inline src/functions.php:195]: its invalid phpdoc to combine `void` with another type

Comment (antonmedv) [reply src/functions.php:195]: This is strange. Why func can't return something in one code path, and not return in another?

Comment (clxmstaab) [reply src/functions.php:195]: its `Task|null` then and requires and explicit `return null`.

thats the opinion of phpstan though. in php-src its not necessary a technical requirement.

**Comment by clxmstaab** [comment]
cannot regenerate the docs, since https://github.com/deployphp/deployer/issues/2253

File: src/functions.php (hunk 1 above)
Comment (antonmedv) [reply src/functions.php:195]: Maybe it’s possible to allow Task|void? Because it’s valid php after all.

Comment (clxmstaab) [reply src/functions.php:195]: feel free to discuss it with the phpstan maintainer :-)

https://github.com/phpstan/phpstan/issues/4014

**Comment by clxmstaab** [comment]
remaining phpstan error is fixed in https://github.com/deployphp/deployer/pull/2260

**Comment by clxmstaab** [comment]
ready to land

**Pull request merged**